public/app/mu-plugins/moj-auth/401.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * A dynamic 401 page.
+ * 
+ * We can either redirect to the login endpoint.
+ * Or, if mamximum attempts have been used, then do nothing.
+ */
+
+namespace MOJ\Intranet;
+
+// Exit if this file is included within a WordPress request.
+if (defined('ABSPATH')) {
+    error_log('moj-auth/401.php was accessed within the context of WordPress.');
+    http_response_code(401) && exit();
+}
+
+define('DOING_STANDALONE_401', true);
+
+$autoload = '../../../../vendor/autoload.php';
+
+if (!file_exists($autoload)) {
+    error_log('moj-auth/401.php autoloader.php was not found.');
+    http_response_code(401) && exit();
+}
+
+require_once  $autoload;
+require_once 'traits/jwt.php';
+require_once 'traits/utils.php';
+
+class Standalone401
+{
+    use AuthJwt;
+    use AuthUtils;
+
+    private $now            = null;
+    private $debug          = false;
+    private $https          = false;
+    private $sub            = '';
+
+    const OAUTH_LOGIN_URI = '/auth/login';
+    const MAX_AUTO_LOGIN_ATTEMPTS = 5;
```

Comment:
This might need to be 1. As we're expecting Entra to work.

Reply:
Set to 1. And set the retry attempts to be logged in a session cookie. User can reset the count to 0 by restating the browser.